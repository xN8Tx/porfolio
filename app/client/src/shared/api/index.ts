export { publicHTTP } from "./public-http";
