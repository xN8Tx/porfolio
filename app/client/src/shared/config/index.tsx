export const availableThemes = ["light", "dark"];
