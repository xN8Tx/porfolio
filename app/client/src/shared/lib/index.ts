export * from "./cn";
export * from "./add-zero-to-num";
