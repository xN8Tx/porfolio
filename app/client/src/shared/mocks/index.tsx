import { render } from "@testing-library/react";
import { Children } from "react";

export const mockStrapiImage = {
  id: 1,
  documentId: "l73ns0nchtgvjgk49ti5vu9c",
  name: "whitecrow.webp",
  alternativeText: null,
  caption: null,
  width: 2012,
  height: 1695,
  hash: "whitecrow_26b66602d4",
  ext: ".webp",
  mime: "image/webp",
  size: 30.67,
  url: "/uploads/whitecrow_26b66602d4.webp",
  previewUrl: null,
  provider: "local",
  provider_metadata: null,
  blurHash:
    "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAEAElEQVR4AcXBQWIbxgEEwZ4l/v9fAjs9AcQodk7JyarK3viDDn/Y4Q87/EEDHvwhAgMOf4DAgAEP/mECAmNs48E/SECGihvbePAPEehERUXHGA/+AZ2oVFFRcWMbD/6n8d/C/2sTHbVUUami4sY2HmX8svHLxnjb+GUj/AiQhPCWAIHwFggw3gYbbqhUUanSlipuTHHjcV8vfhlv42Mb4W3jI/wIEEICSQhvCX+3gRtOVFSqVKnSihOVbTxerxeMt/HLeBvhbeMjQIAkBEhCeEsIbwkf2xiwDZUqKlWqtKWKihvbeLyeT/5jI/zbIIzwI4QEDpCEJHwk4bdtDFBxo0qVKm2pUkXFyQaP1/PJb+Ft4yNAgAABAiRBQgJJCG8Jv21jG26o1FFLlVpuRaWKG9t4vJ5P/i6DMD4CJOEAAZJwgCQk4SP8GDBgGyrdUGnLVWpppUoVJxs8nt/fhL8ECBAgCQEOkIQDmJCEAEn4Oze24UaVKm25SluuUouKDjcer+eT3wIECBAgCQc4CQFOQhIOkISPJIzBYBtudEOllltpy1VuS5UqTrbxeH5/8xF+JOEAAU7CAZJwEk7gEExIQvjLNrbhRidVWrlKW27LVdpSxQ03Hs/vb8JbIIQAJxDCSTiBQzgJJ+EknEAIvwQYDNiGSpUqV2nLbbmWW7mWVtzYxuP1/OYjhAAJHMJJOAkn4QS+EpYwgglJCD8GbGMbKlWqXMut3JbbcluupYoONx7P7ycfARI4hJNwEk7CV8JXwhJOwkk4CeEtEGCDbbihUuUqt+W23Jbb8mqp5So6tvF4fX/zkUAIJ+EkfCWchCXshJPDV8ISTDj8NzdUnLRyldtyW14tt5dXSyvXosONx/P5TYAQknACXzmY8HXCcljC1wnksIQkjLcA4224oaNKLbdyW27L7eV1y225lltxouPxej4JEMJJOAkmfJ1ADpwDCZwDCUs4CQtvAQYDJzqqtKWW23Jvub3cW14tt+VaVNx43OcTCAmchK+Erxw4gRw4B86BE8hhCUtIILwNtuGGSpVabktvub3cW24vt+W23JYqbjzu68XHSVjCEnYO5MA55Bw4h5yQHEggIYHwNhhDh4qWVmxpL72X3nJ7ubfclmup4sbjvi4JSDgJXydwDskh5+DXITvoIUfIgYQEwttgG05UWrGlLe2l99J76S3tpS1tqeLGw14+TgIJ8ZAzPEOGjGx4RnbIGUuAMD7GHE5UVrHFXrzFFlvsxXtpS1uquPGwBQIBcsgZYcgQEDjAgAVGWAKEH2OOTaaoqEyZxRZbbFGxxRYVNx46kiEhyDyMsYxNZhhjjJ2xwfgRBoNtsDEHEyZzbGMb29jGlE22sckc/wJBOGkmYfYa3wAAAABJRU5ErkJggg==",
};
