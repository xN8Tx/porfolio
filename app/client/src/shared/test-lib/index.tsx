import { NextIntlClientProvider } from "next-intl";
import { ThemeProvider } from "next-themes";

import { render as renderRTL } from "@testing-library/react";

import defaultMessages from "../../../messages/en.json";
import { Suspense } from "react";

const Providers = ({ children }: { children: React.ReactNode }) => (
  <Suspense fallback={<></>}>
    <NextIntlClientProvider messages={defaultMessages} locale="en">
      <ThemeProvider defaultTheme="dark">{children}</ThemeProvider>
    </NextIntlClientProvider>
  </Suspense>
);

export const render = (ui: React.ReactElement, options = {}) =>
  renderRTL(ui, { wrapper: Providers, ...options });
