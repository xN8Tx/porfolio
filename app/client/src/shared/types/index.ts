export * from "./api";
export * from "./strapi";
export * from "./next";
