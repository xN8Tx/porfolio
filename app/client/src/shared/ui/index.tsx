export * from "./paragraph";
export * from "./heading";
export * from "./button";
export * from "./image";
export * from "./link";
export * from "./svg";
