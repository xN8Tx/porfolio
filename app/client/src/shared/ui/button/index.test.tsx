import { screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Button } from "./";
import { NextIntlClientProvider } from "next-intl";
import { render } from "@/shared/test-lib";

describe("UI/Button Testing", () => {
  test("should render with _blank target", () => {
    render(
      <Button subtitle={"Subtitle"} href={"/link"} isExternal={true}>
        Children
      </Button>,
    );

    const paragraph = screen.getByTestId("ui-button_paragraph");
    expect(paragraph.textContent).toBe("Children");

    const subtitle = screen.getByTestId("ui-button_subtitle");
    expect(subtitle.textContent).toBe("Subtitle");

    const link = screen.getByTestId<HTMLAnchorElement>("ui-button_link");
    expect(link.href).toBe("http://localhost/en/link");
    expect(link.target).toBe("_blank");
  });

  test("should render with _self target", () => {
    render(
      <NextIntlClientProvider messages={{}} locale="en">
        <Button subtitle={"Subtitle"} href={"/link"} isExternal={false}>
          Children
        </Button>
      </NextIntlClientProvider>,
    );
    const paragraph = screen.getByTestId("ui-button_paragraph");
    expect(paragraph.textContent).toBe("Children");

    const subtitle = screen.getByTestId("ui-button_subtitle");
    expect(subtitle.textContent).toBe("Subtitle");

    const link = screen.getByTestId<HTMLAnchorElement>("ui-button_link");
    expect(link.href).toBe("http://localhost/en/link");
    expect(link.target).toBe("_self");
  });

  test("should correct render small", () => {
    const tree = render(
      <Button
        subtitle={"Subtitle"}
        href={"/link"}
        isExternal={false}
        size="small"
      >
        Children
      </Button>,
    );

    expect(tree).toMatchSnapshot();
  });

  test("should correct render regular", () => {
    const tree = render(
      <Button
        subtitle={"Subtitle"}
        href={"/link"}
        isExternal={false}
        size="regular"
      >
        Children
      </Button>,
    );

    expect(tree).toMatchSnapshot();
  });

  test("should correct render large", () => {
    const tree = render(
      <Button
        subtitle={"Subtitle"}
        href={"/link"}
        isExternal={false}
        size="large"
      >
        Children
      </Button>,
    );

    expect(tree).toMatchSnapshot();
  });

  test("should correct render uppercase", () => {
    const tree = render(
      <Button
        subtitle={"Subtitle"}
        href={"/link"}
        isExternal={false}
        uppercase={true}
      >
        Children
      </Button>,
    );

    expect(tree).toMatchSnapshot();
  });

  test("should correct render light", () => {
    const tree = render(
      <Button
        subtitle={"Subtitle"}
        href={"/link"}
        isExternal={false}
        color="light"
      >
        Children
      </Button>,
    );

    expect(tree).toMatchSnapshot();
  });

  test("should correct render background", () => {
    const tree = render(
      <Button
        subtitle={"Subtitle"}
        href={"/link"}
        isExternal={false}
        color="background"
      >
        Children
      </Button>,
    );

    expect(tree).toMatchSnapshot();
  });

  test("should correct render default", () => {
    const tree = render(
      <Button
        subtitle={"Subtitle"}
        href={"/link"}
        isExternal={false}
        color="default"
      >
        Children
      </Button>,
    );

    expect(tree).toMatchSnapshot();
  });
});
