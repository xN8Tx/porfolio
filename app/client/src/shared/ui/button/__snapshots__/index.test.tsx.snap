// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Button Testing should correct render background 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="ui-link__button hover regular background"
        data-testid="ui-button_button"
      >
        <a
          data-testid="ui-button_link"
          href="/en/link"
          target="_self"
        >
          <p
            data-testid="ui-button_paragraph"
          >
            Children
          </p>
          <p
            data-testid="ui-button_subtitle"
          >
            Subtitle
          </p>
        </a>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="ui-link__button hover regular background"
      data-testid="ui-button_button"
    >
      <a
        data-testid="ui-button_link"
        href="/en/link"
        target="_self"
      >
        <p
          data-testid="ui-button_paragraph"
        >
          Children
        </p>
        <p
          data-testid="ui-button_subtitle"
        >
          Subtitle
        </p>
      </a>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI/Button Testing should correct render default 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="ui-link__button hover regular default"
        data-testid="ui-button_button"
      >
        <a
          data-testid="ui-button_link"
          href="/en/link"
          target="_self"
        >
          <p
            data-testid="ui-button_paragraph"
          >
            Children
          </p>
          <p
            data-testid="ui-button_subtitle"
          >
            Subtitle
          </p>
        </a>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="ui-link__button hover regular default"
      data-testid="ui-button_button"
    >
      <a
        data-testid="ui-button_link"
        href="/en/link"
        target="_self"
      >
        <p
          data-testid="ui-button_paragraph"
        >
          Children
        </p>
        <p
          data-testid="ui-button_subtitle"
        >
          Subtitle
        </p>
      </a>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI/Button Testing should correct render large 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="ui-link__button hover large default"
        data-testid="ui-button_button"
      >
        <a
          data-testid="ui-button_link"
          href="/en/link"
          target="_self"
        >
          <p
            data-testid="ui-button_paragraph"
          >
            Children
          </p>
          <p
            data-testid="ui-button_subtitle"
          >
            Subtitle
          </p>
        </a>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="ui-link__button hover large default"
      data-testid="ui-button_button"
    >
      <a
        data-testid="ui-button_link"
        href="/en/link"
        target="_self"
      >
        <p
          data-testid="ui-button_paragraph"
        >
          Children
        </p>
        <p
          data-testid="ui-button_subtitle"
        >
          Subtitle
        </p>
      </a>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI/Button Testing should correct render light 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="ui-link__button hover regular light"
        data-testid="ui-button_button"
      >
        <a
          data-testid="ui-button_link"
          href="/en/link"
          target="_self"
        >
          <p
            data-testid="ui-button_paragraph"
          >
            Children
          </p>
          <p
            data-testid="ui-button_subtitle"
          >
            Subtitle
          </p>
        </a>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="ui-link__button hover regular light"
      data-testid="ui-button_button"
    >
      <a
        data-testid="ui-button_link"
        href="/en/link"
        target="_self"
      >
        <p
          data-testid="ui-button_paragraph"
        >
          Children
        </p>
        <p
          data-testid="ui-button_subtitle"
        >
          Subtitle
        </p>
      </a>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI/Button Testing should correct render regular 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="ui-link__button hover regular default"
        data-testid="ui-button_button"
      >
        <a
          data-testid="ui-button_link"
          href="/en/link"
          target="_self"
        >
          <p
            data-testid="ui-button_paragraph"
          >
            Children
          </p>
          <p
            data-testid="ui-button_subtitle"
          >
            Subtitle
          </p>
        </a>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="ui-link__button hover regular default"
      data-testid="ui-button_button"
    >
      <a
        data-testid="ui-button_link"
        href="/en/link"
        target="_self"
      >
        <p
          data-testid="ui-button_paragraph"
        >
          Children
        </p>
        <p
          data-testid="ui-button_subtitle"
        >
          Subtitle
        </p>
      </a>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI/Button Testing should correct render small 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="ui-link__button hover small default"
        data-testid="ui-button_button"
      >
        <a
          data-testid="ui-button_link"
          href="/en/link"
          target="_self"
        >
          <p
            data-testid="ui-button_paragraph"
          >
            Children
          </p>
          <p
            data-testid="ui-button_subtitle"
          >
            Subtitle
          </p>
        </a>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="ui-link__button hover small default"
      data-testid="ui-button_button"
    >
      <a
        data-testid="ui-button_link"
        href="/en/link"
        target="_self"
      >
        <p
          data-testid="ui-button_paragraph"
        >
          Children
        </p>
        <p
          data-testid="ui-button_subtitle"
        >
          Subtitle
        </p>
      </a>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI/Button Testing should correct render uppercase 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="ui-link__button hover regular default uppercase"
        data-testid="ui-button_button"
      >
        <a
          data-testid="ui-button_link"
          href="/en/link"
          target="_self"
        >
          <p
            data-testid="ui-button_paragraph"
          >
            Children
          </p>
          <p
            data-testid="ui-button_subtitle"
          >
            Subtitle
          </p>
        </a>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="ui-link__button hover regular default uppercase"
      data-testid="ui-button_button"
    >
      <a
        data-testid="ui-button_link"
        href="/en/link"
        target="_self"
      >
        <p
          data-testid="ui-button_paragraph"
        >
          Children
        </p>
        <p
          data-testid="ui-button_subtitle"
        >
          Subtitle
        </p>
      </a>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
